src/Common/Common.Api/Datasets/DatasetsClient.cs

```diff
@@ -44,40 +45,48 @@ public IEnumerable<Dataset> GetDatasetsAsAdmin(string filter = null, int? top =
         {
-            return this.Client.Datasets.GetDatasetsAsAdmin(groupId: workspaceId.ToString(), filter: filter, top: top, skip: skip).Value?.Select(x => (Dataset)x);
+            return this.Client.Datasets.GetDatasetsAsAdmin(groupId: workspaceId, filter: filter, top: top, skip: skip).Value?.Select(x => (Dataset)x);
         }
 
         public IEnumerable<Datasource> GetDatasources(Guid datasetId, Guid? workspaceId = default)
         {
             var result = workspaceId.HasValue && workspaceId.Value != default ?
-                this.Client.Datasets.GetDatasources(groupId: workspaceId.Value.ToString(), datasetKey: datasetId.ToString()) :
-                this.Client.Datasets.GetDatasources(datasetKey: datasetId.ToString());
+                this.Client.Datasets.GetDatasources(groupId: workspaceId.Value, datasetId: datasetId) :
+                this.Client.Datasets.GetDatasources(datasetId: datasetId);
             return result.Value?.Select(x => (Datasource)x);
 
         }
 
         public IEnumerable<Datasource> GetDatasourcesAsAdmin(Guid datasetId)
         {
-            return this.Client.Datasets.GetDatasourcesAsAdmin(datasetId.ToString()).Value?.Select(x => (Datasource)x);
+            return this.Client.Datasets.GetDatasourcesAsAdmin(datasetId).Value?.Select(x => (Datasource)x);
         }
 
         public IEnumerable<Table> GetTables(Guid datasetId, Guid? workspaceId = default)
         {
             var result = workspaceId.HasValue && workspaceId.Value != default ?
-                this.Client.Datasets.GetTables(groupId: workspaceId.Value.ToString(), datasetKey: datasetId.ToString()) :
-                this.Client.Datasets.GetTables(datasetKey: datasetId.ToString());
+                this.Client.Datasets.GetTables(groupId: workspaceId.Value, datasetId: datasetId) :
+                this.Client.Datasets.GetTables(datasetId: datasetId);
             return result.Value?.Select(x => (Table)x);
         }
 
         public Table UpdateTable(Table table, Guid datasetId, Guid? workspaceId = null)
         {
             var result = workspaceId.HasValue && workspaceId.Value != default ?
-                this.Client.Datasets.PutTableInGroup(groupId: workspaceId.Value.ToString(), datasetKey: datasetId.ToString(), tableName: table.Name, (Microsoft.PowerBI.Api.V2.Models.Table)table) :
-                this.Client.Datasets.PutTable(datasetKey: datasetId.ToString(), tableName: table.Name, (Microsoft.PowerBI.Api.V2.Models.Table)table);
-            return result as Table;
+                this.Client.Datasets.PutTableInGroup(groupId: workspaceId.Value, datasetId: datasetId, tableName: table.Name, (Microsoft.PowerBI.Api.V2.Models.Table)table) :
+                this.Client.Datasets.PutTable(datasetId: datasetId, tableName: table.Name, (Microsoft.PowerBI.Api.V2.Models.Table)table);
+
+            if (result != null)
+            {
+                return (Table)result;
```

Comment:
Could cast using as instead. I think that would help you avoid the if-else return null block.